ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -165,7 +162,8 @@ ExecuteResult execute(final KsqlPlan plan) {
    * @param pullQueryMetrics JMX metrics
    * @return the rows that are the result of evaluating the pull query
    */
-  PullQueryResult executePullQuery(
+  PullQueryResult executeTablePullQuery(
+      final ImmutableAnalysis analysis,
```

Comment:
leaving notes to make sure I understand the refactor. this is the "meat", right? we no longer do the analysis at each part of the query but rather ahead-of-time, which allows us to "understand" whether it's a table pull/stream pull before choosing the corresponding method?

Reply:
Yes!